appinventor/appengine/src/com/google/appinventor/client/explorer/youngandroid/ProjectList.java

```diff
@@ -319,139 +198,182 @@ public void refreshTable(boolean needToSort, boolean isInTrash) {
   }
 
-  /**
-   * Returns true if all projects under the current view have been selected, and false if not.
-   */
-  public boolean isAllProjectsSelected() {
-    if (Ode.getInstance().getCurrentView() == Ode.PROJECTS
-          && getSelectedProjectsCount() == getMyProjectsCount()) {
-      return true;
-    }
-    if (Ode.getInstance().getCurrentView() == Ode.TRASHCAN
-          && getSelectedProjectsCount() == projects.size() - getMyProjectsCount()) {
-      return true;
-    }
-    return false;
+  // FolderManagerEventListener implementation
+  @Override
+  public void onFolderAdded(ProjectFolder folder) {
+    refresh();
   }
-  /**
-   * Returns the list of selected projects.
-   *
-   * @return the selected projects
-   */
-  public List<Project> getSelectedProjects() {
-    return selectedProjects;
+
+  @Override
+  public void onFolderRemoved(ProjectFolder folder) {
+    refresh();
+  }
+
+  @Override
+  public void onFolderRenamed(ProjectFolder folder) {
+    refresh();
+  }
+
+  @Override
+  public void onFoldersChanged() {
+    refresh();
   }
 
-  //  ProjectManagerEventListener implementation
+  @Override
+  public void onFoldersLoaded() {
+    setIsTrash(isTrash);
+  }
 
   @Override
   public void onProjectAdded(Project project) {
-    projects.add(project);
-    projectWidgets.put(project, new ProjectWidgets(project));
-    if (!projectListLoading) {
-      refreshTable(true);
+    if (folder == null) {
+    }
+    if (projectsLoaded) {
+      folder.addProject(project);
+      Ode.getInstance().getFolderManager().saveAllFolders();
+      refresh();
     }
   }
 
   @Override
   public void onTrashProjectRestored(Project project) {
-    selectedProjects.remove(project);
-    refreshTable(false);
-    Ode.getInstance().getProjectToolbar().updateButtons();
+    Ode.getInstance().getFolderManager().getGlobalFolder().addProject(project);
+    Ode.getInstance().getFolderManager().getTrashFolder().removeProject(project);
+    Ode.getInstance().getFolderManager().saveAllFolders();
+    refresh();
   }
 
   @Override
   public void onProjectTrashed(Project project) {
-    selectedProjects.remove(project);
-    refreshTable(false);
-    Ode.getInstance().getProjectToolbar().updateButtons();
+    folder.removeProject(project);
+    Ode.getInstance().getFolderManager().getTrashFolder().addProject(project);
+    Ode.getInstance().getFolderManager().saveAllFolders();
+    refresh();
+  }
+
+  public void onProjectMoved(Project project) {
+    refresh();
   }
 
   @Override
   public void onProjectDeleted(Project project) {
-    projects.remove(project);
-    projectWidgets.remove(project);
-    refreshTable(false);
-    Ode.getInstance().getProjectToolbar().updateButtons();
+    folder.removeProject(project);
+    Ode.getInstance().getFolderManager().saveAllFolders();
+    refresh();
   }
 
   @Override
   public void onProjectsLoaded() {
+    projectsLoaded = true;
     projectListLoading = false;
-    refreshTable(true);
+    refresh();
```

Comment:
This needs to be `refresh(true)` rather than `refresh()` since `refresh()` means `refresh(false)`.